rs/nns/sns-wasm/src/sns_wasm.rs

```diff
@@ -1723,6 +1768,43 @@ where
                 )
             })
     }
+
+    // TODO[NNS1-2997]: Remove this function once the migration is completed.
+    pub fn populate_wasm_metadata(&mut self) {
+        let all_wasm_hashes = self.wasm_indexes.keys().cloned().collect::<Vec<_>>();
+        for hash in all_wasm_hashes {
+            if let Some(index) = self.wasm_indexes.get(&hash) {
+                if index.metadata != vec![] {
+                    // No need to recompute metadata if it's already set. This code is unlikely to
+                    // actually be executed as this is a synchronous migration.
+                    continue;
+                }
+            }
+            let metadata = self
+                .get_wasm_metadata_impl(GetWasmMetadataRequestPb {
```

Comment:
Isn't this supposed to now use read_wasm_metadata_or_err?  B/c get_wasm_metadata_impl now points at the data already in the index?

I think maybe you need to add another test, b/c these probably should have caused a failure somewhere?

Reply:
Ah, yes, you're totally right. This was tested manually, and not the code has evolved in a way that invalidates that test's results. 

Tomorrow, I'll add a test to this branch that would also run on CI (and fix the issue).

Reply:
Fixed and added a test here: https://github.com/dfinity/ic/pull/977/files#diff-2d930ed6199ad023dafa9d3aa47c9c2505fc61ddd028ade4c6b651cd11c253b2R73